File: src/main/kotlin/no/nav/arbeidsgiver/min_side/models/Organisasjon.java
package no.nav.arbeidsgiver.min_side.models;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@JsonIgnoreProperties(ignoreUnknown = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Organisasjon {
    @JsonProperty("Name")
    private String name;
    @JsonProperty("Type")
    private String type;
    @JsonProperty("ParentOrganizationNumber")
    private String parentOrganizationNumber;
    @JsonProperty("OrganizationNumber")
    private String organizationNumber;
    @JsonProperty("OrganizationForm")
    private String organizationForm;
    @JsonProperty("Status")
    private String status;
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/models/Organisasjon.kt
package no.nav.arbeidsgiver.min_side.models

import no.nav.arbeidsgiver.min_side.services.ereg.EregOrganisasjon
import no.nav.arbeidsgiver.min_side.services.ereg.EregOrganisasjon.Companion.orgnummerTilOverenhet
import no.nav.arbeidsgiver.min_side.services.ereg.GyldighetsPeriode.Companion.erGyldig

data class Organisasjon(
    var name: String,
    var parentOrganizationNumber: String? = null,
    var organizationNumber: String,
    var organizationForm: String,
    ) {
    companion object {
        fun fromEregOrganisasjon(eregOrganisasjon: EregOrganisasjon?): Organisasjon? {
            if (eregOrganisasjon == null) {
                return null
            }
            return Organisasjon(
                name = eregOrganisasjon.navn.sammensattnavn,
                parentOrganizationNumber = eregOrganisasjon.orgnummerTilOverenhet(),
                organizationNumber = eregOrganisasjon.organisasjonsnummer,
                organizationForm = eregOrganisasjon.organisasjonDetaljer.enhetstyper?.first { it.gyldighetsperiode.erGyldig() }?.enhetstype ?: ""
            )
        }
    }
}
